cli/internal/env/env.go

```diff
@@ -9,7 +9,55 @@ import (
 	"github.com/vercel/turbo/cli/internal/util"
 )
 
-func getEnvMap() map[string]string {
+// Details contains a map of environment variables broken down by the source
+type Details struct {
+	Explicit EnvironmentVariableMap
+	Prefixed EnvironmentVariableMap
+}
+
+// DetailedMap contains the composite and the detailed maps of environment variables
+// Composite is used as a taskhash input (taskhash.CalculateTaskHash)
+// Details is used to print out a Dry Run Summary
+type DetailedMap struct {
+	Composite EnvironmentVariableMap
+	Details   Details
+}
+
+// EnvironmentVariableMap is a map of env variables and their values
+type EnvironmentVariableMap map[string]string
+
+// Merge takes another EnvironmentVariableMap and merges it into the receiver
+// It overwrites values if they already exist, but since the source of both will be os.Environ()
+// it doesn't matter
+func (evm EnvironmentVariableMap) Merge(another EnvironmentVariableMap) {
+	for k, v := range another {
+		evm[k] = v
+	}
+}
+
+// EnvironmentVariablePairs is a list of "k=v" strings for env variables and their values
+type EnvironmentVariablePairs []string
+
+// ToHashable returns a deterministically sorted set of EnvironmentVariablePairs from an EnvironmentVariableMap
+// This is the value that is used upstream as a task hash input, so we need it to be deterministic
+func (evm EnvironmentVariableMap) ToHashable() EnvironmentVariablePairs {
+	// convert to set to eliminate duplicates
+	unique := make(util.Set, len(evm))
```

Comment:
In _theory_ we don't need this anymore because our source is a map, so all keys will be unique anyway. So we could eliminate this to save a little overhead of converting back and forth.

Reply:
good call, removed